job/server/src/main/java/alluxio/job/util/JobUtils.java

```diff
@@ -147,7 +147,7 @@ public static void loadBlock(URIStatus status, FileSystemContext context, long b
     }
     // TODO(bin): remove the following case when we consolidate tier and medium
     // since there is only one element in the set, we take the first element in the set
-    String medium = pinnedLocation.iterator().next();
+    String medium = pinnedLocation.isEmpty() ? "" : pinnedLocation.iterator().next();
```

Comment:
What is the behavior when medium is empty?

Reply:
medium is actually empty typically. it is non-empty only when the file is pinned to a medium which is rare